### PR TITLE
[JH] Refactor list items from List to ManageList

### PR DESCRIPTION
**Pull request by hsiangj**
## Description

This code separates all list items related logic and UI from List.jsx into ManageList.jsx. In addition, a value attribute was added to the share list email input field to reflect `recipientEmail` useState.  

Note: The adding of the 'Manage List' link in navbar will be handled in a separate issue. Please navigate to the page using '/manage-list' in the URL in the meantime. 

## Related Issue

Sub-issue of #14 

## Acceptance Criteria

- [x] List items are no longer visible in List.jsx
- [x] List items, including add items and share list features, are moved to a separate component

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

### Before
/list (List.jsx) 
Screen Shot 2024-04-01 at 5 19 20 PM

### After
/manage-list (ManageList.jsx)
Screen Shot 2024-04-01 at 5 19 36 PM
/list (List.jsx)
Screen Shot 2024-04-02 at 8 54 06 AM

## Testing Steps / QA Criteria

1. Sign in.
2. Navigate to List page, the list items section should no longer be there.
3. Select a list.
4. Navigate to '/manage-list' in the url. List items should appear and function the same prior to refactoring.

### CODE REVIEW

**Review by piecanoe — CHANGES_REQUESTED** [review]
Nice job separating the list of lists and the list of items!

One thing I noticed--did we remove the redirect routers on the list name buttons from issue #32? That might have been something that happened when the 'List' page and 'Manage List' pages got converged, but wondering if we would we be able to add that back?

**Comment by hsiangj** [comment]
@piecanoe Yes! I noticed that too. I was going to open a different PR to address it but somehow made my current PR disappear. Thanks for bringing it back! 
For the redirect, I believe it's in a different component (SingleList.jsx). I'm wondering if it might be better to keep the PRs separate even though it's a small change?

**Comment by piecanoe** [comment]
Oh no! Glad whatever happened with the change request caught it in time.

But yeah, I think making a separate PR for the redirect totally works. Once you're able to sort out the branches issue, I think this should be good to go.

**Review by piecanoe — APPROVED** [review]

**Review by amalyam — APPROVED** [review]
Works as expected! Thanks for doing the big page shifting on this!

File: src/views/DeprecatedManageList.jsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
+import { useState } from 'react';
+import { Navigate } from 'react-router-dom';
+import { addItem, shareList } from '../api';
+
+export function ManageListDNU({ listPath, currentUserId }) {
```
Comment (mxmason) [inline src/views/DeprecatedManageList.jsx]: 1️⃣ This function should also be renamed to `DeprecatedManageList`, I assume!

2️⃣ For a bit of a small developer experience improvement, you could add a [JSDoc](https://jsdoc.app/about-getting-started) comment to `DeprecatedManageList` to mark it as deprecated.  The comment sits on the lines immediately above the name of the component, and it looks like this:

```js
/**
 * @deprecated
 */
```

This is just for your benefit and the benefit of anyone else reading the code: if a person's editor understands it, it'll decorate the component with a strike-through, showing them at a glance that the component is old and not to be used.

**Review by mxmason — APPROVED** [review]
Good stuff! Left a suggestion.

**Review by krsnamara — APPROVED** [review]
Nice job here. A lot to keep track of in cutting up a big file like this!

**Pull request merged**